am-lib/src/main/java/uk/gov/hmcts/reform/amlib/AccessManagementService.java

```diff
@@ -190,29 +175,26 @@ public FilterResourceResponse filterResource(@NotBlank String userId,
             .build();
     }
 
-    private boolean explicitAccessType(Set<String> userRoles) {
+    private boolean explicitAccessType(String userRole) {
         return jdbi.withExtension(AccessManagementRepository.class,
-            dao -> dao.getRoleAccessType(userRoles.iterator().next())).equals(AccessType.EXPLICIT);
+            dao -> dao.getRoleAccessType(userRole).equals(AccessType.EXPLICIT));
     }
 
     /**
      * Retrieves a list of {@link RoleBasedAccessRecord } and returns attribute and permissions values.
      *
-     * @param serviceName  name of service
-     * @param resourceType type of resource
-     * @param resourceName name of a resource
-     * @param userRoles    A set of user roles
+     * @param resource  {@link ResourceDefinition} a unique service name, resource type and resource name
+     * @param userRoles A set of user roles
      * @return a map of attributes and their corresponding permissions or null
      */
-    public Map<JsonPointer, Set<Permission>> getRolePermissions(@NotBlank String serviceName,
-                                                                @NotBlank String resourceType,
-                                                                @NotBlank String resourceName,
+    public Map<JsonPointer, Set<Permission>> getRolePermissions(@NotNull @Valid ResourceDefinition resource,
                                                                 @NotEmpty Set<@NotBlank String> userRoles) {
         List<Map<JsonPointer, Set<Permission>>> permissionsForRoles =
             jdbi.withExtension(AccessManagementRepository.class, dao ->
                 userRoles.stream()
-                    .map(role -> dao.getRolePermissions(serviceName, resourceType, resourceName, role))
+                    .map(role -> dao.getRolePermissions(resource, role))
                     .map(roleBasedAccessRecords -> roleBasedAccessRecords.stream()
+                        .filter(record -> !explicitAccessType(record.getRoleName()))
```

Comment:
I'd suggest filtering out explicit roles first before calling `dao.getRolePermissions` or making filtering on DB side inside `dao.getRolePermissions` so that if explicit role is passed in method returns empty list.